content/en/docs/concepts/configuration/secret.md

```diff
@@ -15,48 +15,370 @@ weight: 30
+These types vary in terms of the validations performed and the constraints
+Kubernetes imposes on them.
+
+| Builtin Type | Usage |
+|--------------|-------|
+| `Opaque`     |  arbitrary user-defined data |
+| `kubernetes.io/service-account-token` | service account token |
+| `kubernetes.io/dockercfg` | serialized `~/.dockercfg` file |
+| `kubernetes.io/dockerconfigjson` | serialized `~/.docker/config.json` file |
+| `kubernetes.io/basic-auth` | credentials for basic authentication |
+| `kubernetes.io/ssh-auth` | credentials for SSH authentication |
+| `kubernetes.io/tls` | data for a TLS client or server |
+| `bootstrap.kubernetes.io/token` | bootstrap token data |
+
+You can define and use your own Secret type by assigning non-empty string as the
+`type` value for a Secret object. An empty string is treated as `Opaque`.
+Kubernetes doesn't impose any contstraints on the type name you choose. But if
+you are using one of the builtin types listed above, you will have to meet all
+the requirements defined for that type.
+
+#### Opaque Secrets
+
+`Opaque` is the default Secret type if omitted from a Secret configuration file.
+When you create a Secret from the `kubectl` command line, it corresponds to
+the `generic` subcommand. For example, the following command creates an empty
+Secret of type `Opaque`.
 
-### Built-in Secrets
+```shell
+kubectl create secret generic empty-secret
+kubectl get secret empty-secret
+```
 
-#### Service accounts automatically create and attach Secrets with API credentials
+The output looks like:
+
+```
+NAME           TYPE     DATA   AGE
+empty-secret   Opaque   0      2m6s
+```
+
+The `DATA` column shows the number of data items stored in the Secret.
+In this case, `0` means we have just created an empty Secret.
+
+####  Service account token Secrets
+
+A `kubernetes.io/service-account-token` type of Secret is used to store a
+token that identifies a service acccount. When using this Secret type, you need
+to ensure that the `kubernetes.io/service-account.name` annotation is set to an
+existing service account name. An Kubernetes controller will fill in some other
+fields such as the `kubernetes.io/service-account.uid` annotation and the
+`token` key in the `data` field set to actual token content.
+
+The following YAML is an example config for a service account token Secret:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: secret-sa-sample
+  annotations:
+    kubernetes.io/service-account.name: "sa-name"
+type: kubernetes.io/service-account-token
+data:
+  # You can include additional key value pairs as you do with Opaque Secrets
+  extra: YmFyCg==
+```
 
-Kubernetes automatically creates secrets which contain credentials for
-accessing the API and automatically modifies your Pods to use this type of
-secret.
+In fact, when creating a Pod, Kubernetes automatically creates a service
+account Secret which contain credentials for accessing the API and
```

Comment:
nit: could simplify
When creating a Pod, Kubernetes automatically creates a service account Secret and
automatically modifies your Pod to use this secret. The service account Secret contains
credentials for accessing the API.